community/io/src/main/java/org/neo4j/io/pagecache/PagedFile.java

```diff
@@ -159,4 +163,30 @@
      * @see AutoCloseable#close()
      */
     void close() throws IOException;
+
+    /**
+     * Open a {@link ReadableByteChannel} view of this PagedFile.
+     * <p>
+     * The channel will read the file sequentially from the beginning till the end.
+     * Seeking is not supported.
+     * <p>
+     * The channel is not thread-safe.
+     *
+     * @return A channel for reading the paged file.
+     */
+    ReadableByteChannel openReadableByteChannel() throws IOException;
+
+    /**
+     * Open a {@link WritableByteChannel} view of this PagedFile.
+     * <p>
+     * The channel will write to the file sequentially from the beginning of the file, overwriting whatever is there
```

Comment:
At the moment we only support truncating files when mapping them. I imagine `StoreCopyClient` will use that functionality to prevent this from being a problem in practice.